src/runtime_src/core/edge/skd/sk_daemon.cpp

```diff
@@ -38,19 +40,36 @@ unsigned int getHostBO(unsigned long paddr, size_t size)
   return boHandle;
 }
 
-void *mapBO(unsigned int boHandle, bool write)
+static void *mapBO(unsigned int boHandle, bool write)
 {
   void *buf;
   buf = xclMapBO(devHdl, boHandle, write);
   return buf;
 }
 
-void freeBO(unsigned int boHandle)
+static void freeBO(unsigned int boHandle)
 {
   xclFreeBO(devHdl, boHandle);
 }
 
-int getBufferFd(unsigned int boHandle)
+static int logMsg(xrtLogMsgLevel level, const char* tag,
+		       const char* format, ...)
+{
+  static auto verbosity = xrt_core::config::get_verbosity();
+  if (level > verbosity) {
+    return 0;
+  }
+  va_list args;
+  va_start(args, format);
+  int ret = -1;
+  ZYNQ::shim *drv = ZYNQ::shim::handleCheck(devHdl);
+  ret = drv ? drv->xclLogMsg(level, tag, format, args) : -ENODEV;
+  va_end(args);
+
+  return ret;
```

Comment:
How about consolidating all these logMsg functions into xrt_core::message (core/common/message.h)?  Add the missing va_list function there.   

shim::xclLogMsg should never have been a member function of the shim class in the first place.  You can ignore this comment now, since it is a little bit out of scope.  But this is one thing we should change at some point.

Reply:
Agree. Will look into this separately